### DISCUSSION TITLE
How to install and use?

### DISCUSSION THREAD

**Discussion by camya** [squirephp/squire#13] [General]
Hi,

at the moment I try to figure out how to use the project in my application. I followed the instructions by installing squire via `composer require squirephp/countries-en` 

The instruction says, that I can access the data for example by simply using `Country::all();` I've tried it from my Controller.

Unfortunately this does not work for me. I get the following error message.

```
Illuminate\Database\QueryException 
could not find driver (SQL: select * from "countries")
```

It looks like I missed something essential. :) How can setup the driver or how can I migrate the data for the model?

Thank you for your help.

**Comment by danharrin** [comment]
Hey @camya! Could you please post the full controller class that contains this code?

**Comment by camya** [comment]
Hi Dan, nice to meet you.

I output the data in a regular (resouce) controller's index action. Nothing fancy here. Hope I'm not just missing a simple step. :)

```
<?php

namespace App\Http\Controllers;

use Squire\Models\Country;

class ClientsController extends Controller
{

    public function index()
    {
        $countries = Country::all();
        dd($countries);
    }

}
```

 The files installed in vendor/squirephp are looking fine for me.

![vendor](https://user-images.githubusercontent.com/827658/104772174-4b3bd500-5773-11eb-80c4-3a7dc089d8e1.jpg)

**Comment by danharrin** [comment]
Thanks! Could you double check that you have the `sqlite3` and `pdo_sqlite` PHP extensions installed? To do this, you could `dd(extension_loaded('sqlite3'), extension_loaded('pdo_sqlite'));` and check that it outputs `true`, `true`.

**Reply by camya** [reply]
Hi. sqlite wasn't installed on my dev system. I now installed both sqlite3 and pdo_sqlite extensions. The dd(extension_loaded('sqlite3'), extension_loaded('pdo_sqlite')); returned true, true ;)

```
SQLite3::version() [
  "versionString" => "3.30.1"
  "versionNumber" => 3030001
]
```

Now the exception changed a little bit. :) Is there a migration step I miss?

```
Illuminate\Database\QueryException
SQLSTATE[HY000]: General error: 1 no such table: countries (SQL: select * from "countries")
```

**Comment by camya** [comment]
Ok, I found the problem.

The method `boot()` inside `Squire/Model` contains the condition `if(static::hasValidCache())`. This retured true for a cached database file `storage/framework/cache/squire-squire-models-country-en.sqlite` which was created. Unfortunately this file was complete empty. 

I'm not sure why this empy file was created, and why it was empty.

I deleted the squire-squire-models-country-en.sqlite file and reloaded the page. This time a valid database squire-squire-models-country-en.sqlite file was created and Country::all() worked just fine.
 
Some ideas:

- Hint in the docs, that sqlite3 or pdo_sqlite is required.
- A Composer "require" entry for the SQLite extension.
- An Artisan command like "php artisan squire:status" or "php artisan squire:init --model=countries-en --hard|soft" may be useful.

**Reply by Sjoerd24** [reply]
I had the same problem when i made a clone of filament-demo project and tried immediatly switching to mysql database. For some reason the cache database files for squirephp were also empty with me and deleting them and refreshing the page fixed everything, thanks! php artisan squire:cache comment would be helpfull! ;)

**Reply by ShojibFlamon** [reply]
One of the reason of creating empty file might be php memory limit exception

`Allowed memory size of 134217728 bytes exhausted (tried to allocate 4096 bytes)`

So, you can check the php memory limit from php.ini or 
Type in terminal:

`php -r "echo ini_get('memory_limit') . PHP_EOL;"`
or
`grep memory_limit /etc/php/8.2/fpm/php.ini`
or
`php-fpm8.2 -i | grep memory_limit` if you are using php-fpm
or 
check from you php code
`<?php echo phpinfo(); ?>`
![image](https://github.com/user-attachments/assets/90917606-1780-4c80-8713-66f9f64a8716)

Thanks

**Comment by danharrin** [comment]
Awesome! I will add those requirements to the docs - I presumed that they were default extensions since most environments preinstall them by default. How would you feel about a `php artisan squire:cache` command to clear the existing cache and recache all models?

**Reply by camya** [reply]
`php artisan squire:cache` sounds fine to me too. :)

**Comment by superDuperCyberTechno** [comment]
It seems to happen when you load the app without the necessary PHP modules. The file is created regardless of it being populated.

I'm assuming that removing the file when this happens would fix it.

**Comment by waqasra2022skipq** [comment]
This issue still happens. Same steps(deleting the sqlite file in Storage) does worked for me. But It should not have happened at all. Or at least this should have been added to Docs

**Comment by ShaungBhone** [comment]
Thank you and reminder for `php artisan squire:cache`
